selene-lib/src/rules/high_cyclomatic_complexity.rs

```diff
@@ -0,0 +1,389 @@
+use super::*;
+use crate::ast_util::range;
+use std::convert::Infallible;
+
+use full_moon::{
+    ast::{self, Ast},
+    visitors::Visitor,
+};
+
+use serde::Deserialize;
+
+#[derive(Clone, Copy, Deserialize)]
+pub struct HighCyclomaticComplexityConfig {
+    maximum_complexity: u16,
+}
+
+impl Default for HighCyclomaticComplexityConfig {
+    fn default() -> Self {
+        Self {
+            // eslint defaults to 20, but testing on OSS Lua shows that 20 is too aggressive
+            maximum_complexity: 40,
+        }
+    }
+}
+
+#[derive(Default)]
+pub struct HighCyclomaticComplexityLint {
+    config: HighCyclomaticComplexityConfig,
+}
+
+impl Rule for HighCyclomaticComplexityLint {
+    type Config = HighCyclomaticComplexityConfig;
+    type Error = Infallible;
+
+    const SEVERITY: Severity = Severity::Allow;
+    const RULE_TYPE: RuleType = RuleType::Style;
+
+    fn new(config: Self::Config) -> Result<Self, Self::Error> {
+        Ok(HighCyclomaticComplexityLint { config })
+    }
+
+    fn pass(&self, ast: &Ast, _: &Context, _: &AstContext) -> Vec<Diagnostic> {
+        let mut visitor = HighCyclomaticComplexityVisitor {
+            positions: Vec::new(),
+            config: self.config,
+        };
+
+        visitor.visit_ast(ast);
+
+        visitor
+            .positions
+            .into_iter()
+            .map(|(position, complexity)| {
+                Diagnostic::new(
+                    "limit_function_complexity",
```

Comment:
Looks like this is using the old name
```suggestion
                    "high_cyclomatic_complexity",
```